src/ctor.rs

```diff
@@ -229,7 +229,9 @@ where
   F: FnOnce() -> Result<T, E>,
 {
   unsafe {
-    from_placement_try_fn(|mut dest| Ok(dest.set(MaybeUninit::new(f()?))))
+    from_placement_try_fn(|mut dest| {
+      f().map(|val| dest.set(MaybeUninit::new(val)))
```

Comment:
Clippy complains about this. Just a workaround.

```
error: passing a unit value to a function
   --> src/ctor.rs:232:38
    |
232 |     from_placement_try_fn(|mut dest| Ok(dest.set(MaybeUninit::new(f()?))))
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
note: the lint level is defined here
   --> src/lib.rs:146:9
    |
146 | #![deny(warnings, missing_docs, unused)]
    |         ^^^^^^^^
    = note: `#[deny(clippy::unit_arg)]` implied by `#[deny(warnings)]`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unit_arg
help: move the expression in front of the call and replace it with the unit literal `()`
    |
232 |     from_placement_try_fn(|mut dest| {
233 |         dest.set(MaybeUninit::new(f()?));
234 |         Ok(())
235 |     })
    |

error: aborting due to previous error
```

Reply:
WAI. Same philosophy as `-Werror`. Everyone ignores warnings unless they're made into errors.

(I actually think Clippy's rewriting is much clearer, so I'd go with that.)

Reply:
> Same philosophy as `-Werror`.

`-Werror` (`-D warnings`) is fine when you pass it to the compiler (on CI or whatever). The problem is: adding a new warning is not a breaking change. So, if you release a lib with `#![deny(warnings)]`, then compiler updates, introducing new warning which is triggered by your code, then users won't be able to compile your lib!

Reply:
> So, if you release a lib with #![deny(warnings)], then compiler updates, introducing new warning which is triggered by your code, then users won't be able to compile your lib!

I'm aware. This is, again, an intentional choice. =)

Reply:
But why? You can prevent bad code by a simple CI job instead of making break changes.

https://github.com/actions-rs/example/blob/23ffb1bf0016f41999902ba7542b4f1bb1a89c48/.github/workflows/quickstart.yml#L79